src/index.js

```diff
@@ -0,0 +1,404 @@
+    ? ( isError ? '✘' : '✓' )
+    : ''
+
+  const style = s =>
+    isCancel  ? CANCEL_STYLE
+      : isError     ? ERROR_STYLE
+      : s
+
+  const formatter = logFormatter()
+
+  if(winnerInd)
+    formatter.add(`%c ${winnerInd}`, style(LABEL_STYLE))
+
+  if(effect && effect.label)
+    formatter.add(`%c ${effect.label}: `,  style(LABEL_STYLE))
+
+  if(type)
+    formatter.add(`%c ${type} `, style(EFFECT_TYPE_STYLE))
+
+  formatter.add('%c', style(DEFAULT_STYLE))
+
+  return {
+    method,
+    formatter
+  }
+}
+
+function argToString(arg) {
+  return (
+    typeof arg === 'function' ? `${arg.name}`
+      : typeof arg === 'string'   ? `'${arg}'`
+      : arg
+  )
+}
+
+function logResult({status, result, error, duration}, formatter, ignoreResult) {
+
+  if(status === RESOLVED && !ignoreResult) {
+    if( is.array(result) ) {
+      formatter.addValue(' → ')
+      formatter.addValue(result)
+    } else
+      formatter.appendData('→',result)
+  }
+
+  else if(status === REJECTED) {
+    formatter.appendData('→ ⚠', error)
+  }
+
+  else if(status === PENDING)
+    formatter.appendData('⌛')
+
+  else if(status === CANCELLED)
+    formatter.appendData('→ Cancelled!')
+
+  if(status !== PENDING)
+    formatter.appendData(`(${duration.toFixed(2)}ms)`)
+}
+
+function logFormatter() {
+  const logs = []
+  let suffix = []
+
+  function add(msg, ...args) {
+    // Remove the `%c` CSS styling that is not supported by the Node console.
+    if(!IS_BROWSER && typeof msg === 'string') {
+      const prevMsg = msg
+      msg = msg.replace(/^%c\s*/, '')
+      if(msg !== prevMsg) {
+        // Remove the first argument which is the CSS style string.
+        args.shift()
+      }
+    }
+    logs.push({msg, args})
+  }
+
+  function appendData(...data) {
+    suffix = suffix.concat(data)
+  }
+
+  function addValue(value) {
+    if(isPrimitive(value))
+      add(value)
+    else {
+      // The browser console supports `%O`, the Node console does not.
+      if(IS_BROWSER)
+        add('%O', value)
+      else
+        add('%s', inspect(value))
+    }
+  }
+
+  function addCall(name, args) {
+    if(!args.length)
+      add( `${name}()` )
+    else {
+      add(name)
+      add('(')
+      args.forEach( (arg, i) => {
+        addValue( argToString(arg) )
+        addValue( i === args.length - 1 ? ')' : ', ')
+      })
+    }
+  }
+
+  function getLog() {
+    const msgs = []
+    let msgsArgs = []
+    for (var i = 0; i < logs.length; i++) {
+      msgs.push(logs[i].msg)
+      msgsArgs = msgsArgs.concat(logs[i].args)
+    }
+    return [msgs.join('')].concat(msgsArgs).concat(suffix)
+  }
+
+  return {
+    add, addValue, addCall, appendData, getLog
+  }
+}
+
+const logSaga = () => {
+  console.log('')
+  console.log('Saga monitor:', Date.now(), (new Date()).toISOString())
+  logEffectTree(0)
+  console.log('')
+}
+
+// Export the snapshot-logging function to run from the browser console or extensions.
+if(IS_BROWSER) {
+  window.$$LogSagas = logSaga
```

Comment:
Please choose a more unique global name, such as `__REDUX_SAGA_MONITOR_LOG_SAGAS__ `. Please see the reasoning here: https://github.com/zalmoxisus/redux-devtools-extension/issues/202

If you're in doubt that the long name would be difficult to run from the browser developer tools console, the console has got a decent auto-complete and command history, so you will only need to enter (read: copy-paste) that once.